commit 937c7245f0f4058d2b335d160617c1cdb2398cb7
Author: Andriy Rysin <arysin@gmail.com>
Date:   Sat Apr 2 22:27:37 2011 -0400

    Fix null pointer crash when no rules found; add unit test
    BUG: 269961

diff --git a/kcontrol/keyboard/flags.cpp b/kcontrol/keyboard/flags.cpp
index 24cd457..be70b1c 100644
--- a/kcontrol/keyboard/flags.cpp
+++ b/kcontrol/keyboard/flags.cpp
@@ -116,7 +116,7 @@ static QString getDisplayText(const QString& layout, const QString& variant, con
 {
 	if( variant.isEmpty() )
 		return layout;
-	if( rules->version == "1.0" )
+	if( rules == NULL || rules->version == "1.0" )
 		return i18nc("layout - variant", "%1 - %2", layout, variant);
 	return variant;
 }
diff --git a/kcontrol/keyboard/tests/flags_test.cpp b/kcontrol/keyboard/tests/flags_test.cpp
index 0f84ce5..f4cec94 100644
--- a/kcontrol/keyboard/tests/flags_test.cpp
+++ b/kcontrol/keyboard/tests/flags_test.cpp
@@ -64,6 +64,9 @@ private Q_SLOTS:
         QVERIFY( flags->getLongText(layoutUnit1, rules).startsWith("USA - International") );
         QCOMPARE( flags->getLongText(layoutUnit2, rules), QString("USA - other") );
 
+        rules = NULL; // when no rules found
+        QCOMPARE( flags->getLongText(layoutUnit1, rules), QString("us - intl") );
+
         flags->clearCache();
     }
 
